src/main/scala/rocket/Multiplier.scala

```diff
@@ -6,10 +6,12 @@ package freechips.rocketchip.rocket
 import chisel3._
 import chisel3.util.{Cat, log2Up, log2Ceil, log2Floor, Log2, Decoupled, Enum, Fill, Valid, Pipe}
 import freechips.rocketchip.util._
-import ALU._
+import freechips.rocketchip.config.Parameters
 
-class MultiplierReq(dataBits: Int, tagBits: Int) extends Bundle {
-  val fn = Bits(SZ_ALU_FN.W)
+class MultiplierReq(dataBits: Int, tagBits: Int)(implicit val p: Parameters) extends Bundle with HasRocketCoreParameters {
+  val alu = if (usingABLU) ABLU else ALU
```

Comment:
I don't want to make this depend on `HasRocketCoreParameters`.

Reply:
I also do not want to do so. However, Multipler internal staff depends on the rocket ALU and its specific FN encoding. Unless we refactor IDecode to include the decode table defined in Multipler, we have to depend on the rocket core parameters.